common/key/key.go

```diff
@@ -15,63 +15,58 @@ type ExchangePairAsset struct {
 	Asset    asset.Item
 }
 
-// ExchangeAsset is a unique map key signature for exchange and asset
-type ExchangeAsset struct {
-	Exchange string
-	Asset    asset.Item
-}
-
-// PairAsset is a unique map key signature for currency pair and asset
-type PairAsset struct {
-	Base  *currency.Item
-	Quote *currency.Item
-	Asset asset.Item
-}
-
-// SubAccountAsset is a unique map key signature for subaccount and asset
-type SubAccountAsset struct {
-	SubAccount string
-	Asset      asset.Item
-}
-
-// Pair combines the base and quote into a pair
-func (k *PairAsset) Pair() currency.Pair {
-	if k == nil || (k.Base == nil && k.Quote == nil) {
-		return currency.EMPTYPAIR
+// NewExchangePairAssetKey is a helper to reduce the amount of code needed to create a key
+func NewExchangePairAssetKey(exch string, a asset.Item, cp currency.Pair) ExchangePairAsset {
+	return ExchangePairAsset{
+		Exchange: strings.ToLower(exch),
```

Comment:
suggestion: I would leave this out and just pass it directly, because this adds in overhead and can be adjusted by the caller.

Reply:
I meat to get rid of it everywhere for you sorry, will rm